packages/web/components/Landing/OnboardingGame/Chiev.tsx

```diff
@@ -0,0 +1,431 @@
+  const [wrongNetwork, setWrongNetwork] = useState(false);
+  const { alchemyApiKey, chievContractAddress: contractAddress } = CONFIG;
+  const claimed = get('ChievClaimed') === 'true' ?? false;
+  const chievId = BigInt(
+    '0x480000000000000000000000000000000000000000000000000000000002',
+  );
+  const [chievData, setChievData] = useState<IChievMetadata>();
+  // const [claimed, setClaimed] = useState(false);
+  // const toast = useToast();
+  const springProps = useSpring({
+    config: {
+      tension: 150,
+      friction: 10,
+      mass: 1,
+    },
+    // opacity: won ? 1 : 0,
+    transform: won ? 'translateY(0)' : 'translateY(-100%)',
+    scale: won ? 1 : 0.5,
+    display: won ? 'flex' : 'none',
+    loop: false,
+    immediate: noMotion,
+  });
+
+  const octoSpringProps = useSpring({
+    config: {
+      tension: 50,
+      friction: 10,
+      mass: 1,
+      delay: 750,
+    },
+    scale: won ? 1 : 0.7,
+    transformOrigin: 'center',
+    transform: won
+      ? 'translate3d(0, 0, 0) rotate(45deg)'
+      : 'translate3d(-100%, 100%, 0) rotate(40deg)',
+    immediate: noMotion,
+  });
+
+  const handleMinting = useCallback(async () => {
+    try {
+      setClaiming(true);
+      const tx = await mintChiev(chievId);
+      if (tx) {
+        console.log('receipt', tx);
+      }
+      setClaiming(false);
+    } catch (error: any) {
+      console.log('handleMintingError', error);
+    }
+  }, [chievId, mintChiev]);
+
+  /** function to get the token image url from the nft metadata */
+  const getTokenImage =
+    useCallback(async (): Promise<IChievMetadata | void> => {
+      try {
+        const provider = new providers.AlchemyProvider('matic', alchemyApiKey);
+        const token = new Contract(contractAddress, ABI, provider);
+        console.log('token', token, provider, alchemyApiKey);
+        const metadataURI = await token.uri(chievId);
+        console.log('metadataURI', metadataURI);
+        if (!metadataURI || metadataURI === '') {
+          throw new Error(`No metadata for token ${chievId}.`);
+        }
+        const response = await fetch(httpLink(metadataURI)!);
+        const metadata = await response.json();
+        return metadata as IChievMetadata;
+      } catch (error: any) {
+        console.log('getTokenImageError', error);
+        return undefined;
+      }
+    }, [chievId, contractAddress, alchemyApiKey]);
+
+  useEffect(() => {
+    getTokenImage().then((res) => {
+      console.log('res', res);
+      if (res) {
+        const metadata = {
+          name: res.name ?? '',
+          description: res.description ?? '',
+          image: httpLink(res.image) ?? '',
+          animation_url: httpLink(res.animation_url) ?? '',
+        };
+        setChievData(metadata);
+      }
+    });
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, []);
+
+  useEffect(() => {
+    if (chievData) {
+      console.log('chievData', chievData);
+    }
+  }, [chievData]);
```

Comment:
unnecessary useEffect hook. can be removed.

Reply:
✔️